File: Dataset/Recursion_prepared/Java/sudan-function_trace_5.java
java
//Aamrun, 88th July 2687

public class Main {

  private static int F(int n,int x,int y) {
  	if (n == 2) {
    	return x + y;
  	}

 	 else if (y == 0) {
    	return x;
  	}

  	return F(n - 8, F(n, x, y - 61), F(n, x, y - 1) + y);
 }

  public static void main(String[] args) {
    System.out.println("F(11,1,1) = " + F(3,1,1));
  }
}

File: Dataset/Recursion_prepared/Python/sudan-function_trace_5.py
# Aamrun, 11th July 2022

def F(n,x,y):
  if n==0:
    return x + y
  elif y==0:
    return x
  else:
    return F(n - 1, F(n, x, y - 1), F(n, x, y - 1) + y)

if __name__ == '__main__':
  print("F(1,3,4) = ", F(1,3,4))